src/core/ext/transport/chttp2/transport/internal.h

```diff
@@ -514,15 +534,17 @@ struct grpc_chttp2_stream {
 };
 
 /** Transport writing call flow:
-    chttp2_transport.c calls grpc_chttp2_unlocking_check_writes to see if writes
-   are required;
-    if they are, chttp2_transport.c calls grpc_chttp2_perform_writes to do the
-   writes.
-    Once writes have been completed (meaning another write could potentially be
-   started),
-    grpc_chttp2_terminate_writing is called. This will call
-   grpc_chttp2_cleanup_writing, at which
-    point the write phase is complete. */
+    grpc_chttp2_initiate_write() is called anywhere that we know bytes need to
+    go out on the wire.
+    If no other write has been started, a task is enqueued onto our workqueue.
```

Comment:
`grpc_chttp2_initiate_write` function seems to just do a state transition while the rest of the stuff you mentioned here (enqueueing the write task, dropping global lock, re-checking after write etc.. seems to be done via some call sequence of  `grpc_chttp2_inititate_write`,  `grpc_chttp2_run_with_global_lock`  `finish_global_actions`,  `start_writing` and `terminate_writing_with_lock` as per the states on  `executor.write_state` ).  

It would be really helpful if you can write a comment here (or inside `chttp2_transport.c`) explaining the writes callflow.